docs/proposals/20201019-prometheus-rules-in-observatorium-api.md

```diff
@@ -0,0 +1,130 @@
+## Implement API for managing Prometheus Recording Rules
+
+* **Owners:**:
+  * `@squat` / `@ianbillett`
+
+* **Related Tickets:**
+  * `<JIRA, GH Issues>`
+
+* **Other docs:**
+  * Original [Prometheus Rules in Observatorium API design doc](https://docs.google.com/document/d/1F9Cw6I4qFs__0Dcm19xvxJqRBCAGQBBggYg_PQSV_-g/edit#heading=h.cp0jmcyfj3) (internal Red Hat link)
+
+## TL;DR
+
+We propose implementing a multi-tenant API that allows users to create, read, update and delete Prometheus recording rules.
```

Comment:
https://github.com/observatorium/observatorium/pull/419/files#diff-8de1f9b976bdab7fdd01a236d8fed2c2be98ee58ea965d33567ba147fe07479eR45 mentions both `alerting` and `recording` rules. But the title and here only mentions `recording` rules, so is `alerting` rule also included in the proposal?

Reply:
Great question @yeya24 :raised_hands:

The original design doc included both alerting and recording rules in scope. That makes sense given that recording and alerting rules can be freely mixed in a given `rule_files`. 

However, without a corresponding set of features to support user-supplied alertmanager routing configuration instances (which is definitely out of scope of this document) - any alerting rules we ingest will sit idle.

My preference here is to scope this design doc to recording rules only, and explicitly reject any rule files that contain `alert:` keys with an appropriately clear error message to the caller.

Reply:
I've added this to the non-goals section. Resolving.